src/components/Map/MapContextProvider.tsx

```diff
@@ -15,7 +16,14 @@ interface MapContextProviderProps {
 const MapContextProvider = ({ children }: MapContextProviderProps) => {
   const [map, setMap] = useState<Leaflet.Map | undefined>(undefined)
 
-  return <MapContext.Provider value={{ map, setMap }}>{children}</MapContext.Provider>
+  const centerCard = useCallback((lat: number, lng: number, zoom: number) => {
+    console.log("Centering map to:", lat, lng, "with zoom:", zoom);
+    if (map) {
+      map.flyTo([lat, lng], zoom);
+    }
+  }, [map]);
+
+  return <MapContext.Provider value={{ map, setMap, centerCard }}>{children}</MapContext.Provider>
```

Comment:
map instance already has function FlyTo, which we can call directly from the map instance